jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -154,7 +154,7 @@ ecma_builtin_number_prototype_helper_round (lit_utf8_byte_t *digits_p, /**< [in,
     int i = 1;
 
     /* Handle curry number. */
-    for (; i < (int) num_digits; i++)
+    for (; i < (int) num_digits && i <= round_num; i++)
```

Comment:
`round_num` will always be less than `num_digits`, because of the `if` on line 145, so the condition for `num_digits` is unnecessary.